yt_dlp/extractor/goodgame.py

```diff
@@ -9,16 +9,16 @@
 
 class GoodGameIE(InfoExtractor):
     IE_NAME = 'goodgame:stream'
-    _VALID_URL = r'https?://goodgame\.ru/channel/(?P<id>\w+)'
+    _VALID_URL = r'https?://goodgame\.ru/(?:channel/)?(?P<id>[\w.*-]+)'
```

Comment:
Okay, my bad we actually need to follow the redirects as channels can differ from the user
```suggestion
    _VALID_URL = r'https?://goodgame\.ru/(?P<id>[\w.*-]+)'
```
`https://goodgame.ru/channel/JosephStalin` -> `https://goodgame.ru/TGW`